syft/pkg/cataloger/nix/parse_nix_store_path_test.go

```diff
@@ -0,0 +1,284 @@
+package nix
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_findVersionIsh(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		wantIdx int
+		wantEx  string
+		wantPr  string
+	}{
+		{
+			name:    "no version",
+			input:   "5q7vxm9lc4b9hifc3br4sr8dy7f2h0qa-source",
+			wantIdx: -1,
+			wantEx:  "",
+			wantPr:  "",
+		},
+		{
+			name:    "semver with overbite into output",
+			input:   "/nix/store/h0cnbmfcn93xm5dg2x27ixhag1cwndga-glibc-2.34-210-bin",
+			wantIdx: 50,
+			wantEx:  "2.34-210-bin",
+			wantPr:  "210-bin",
+		},
+		{
+			name:    "multiple versions",
+			input:   "5zzrvdmlkc5rh3k5862krd3wfb3pqhyf-perl5.34.1-TimeDate-2.33",
+			wantIdx: 53,
+			wantEx:  "2.33",
+			wantPr:  "",
```

Comment:
Is there any checking of the package name that gets extracted? This is TimeDate at version 2.33, in a perl 5.34.1 environment.

Reply:
Ah, it's further down. Maybe this test function should have a comment at the top, stating that the names themselves are tested in `Test_parseNixStorePath(` further down.